app/products/calls/components/channel_info_start/channel_info_start_button.tsx

```diff
@@ -30,33 +55,59 @@ const ChannelInfoStartButton = ({
     limitRestrictedInfo,
 }: Props) => {
     const intl = useIntl();
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
     const isLimitRestricted = limitRestrictedInfo.limitRestricted;
+    const [connecting, setConnecting] = useState(false);
+    const [joiningMsg, setJoiningMsg] = useState('');
+
+    const starting = intl.formatMessage({id: 'mobile.calls_starting', defaultMessage: 'Starting...'});
+    const joining = intl.formatMessage({id: 'mobile.calls_joining', defaultMessage: 'Joining...'});
 
-    const toggleJoinLeave = useCallback(() => {
+    const toggleJoinLeave = useCallback(async () => {
         if (alreadyInCall) {
             leaveCall();
+            dismissChannelInfo();
         } else if (isLimitRestricted) {
             showLimitRestrictedAlert(limitRestrictedInfo, intl);
+            dismissChannelInfo();
         } else {
-            leaveAndJoinWithAlert(intl, serverUrl, channelId);
-        }
+            setJoiningMsg(isACallInCurrentChannel ? joining : starting);
+            setConnecting(true);
 
-        dismissChannelInfo();
+            await leaveAndJoinWithAlert(intl, serverUrl, channelId);
+            setConnecting(false);
+            dismissChannelInfo();
+        }
     }, [isLimitRestricted, alreadyInCall, dismissChannelInfo, intl, serverUrl, channelId, isACallInCurrentChannel]);
 
     const [tryJoin, msgPostfix] = useTryCallsFunction(toggleJoinLeave);
 
     const joinText = intl.formatMessage({id: 'mobile.calls_join_call', defaultMessage: 'Join call'});
     const startText = intl.formatMessage({id: 'mobile.calls_start_call', defaultMessage: 'Start call'});
     const leaveText = intl.formatMessage({id: 'mobile.calls_leave_call', defaultMessage: 'Leave call'});
+    const text = isACallInCurrentChannel ? joinText + msgPostfix : startText + msgPostfix;
+    const icon = isACallInCurrentChannel ? 'phone-in-talk' : 'phone';
+
+    if (connecting) {
+        return (
+            <Loading
+                color={theme.buttonBg}
+                size={'small'}
+                footerText={joiningMsg}
+                containerStyle={styles.container}
+                footerTextStyles={styles.text}
+            />
+        );
+    }
 
     return (
         <OptionBox
             onPress={preventDoubleTap(tryJoin)}
-            text={startText + msgPostfix}
-            iconName='phone'
-            activeText={joinText + msgPostfix}
-            activeIconName='phone-in-talk'
+            text={text}
+            iconName={icon}
+            activeText={isACallInCurrentChannel ? text : starting}
```

Comment:
This change (and lines 89-90, and the `joiningMsg` and `setJoiningMsg` state) fixes a tricky issue that was plaguing this button -- after pressing, the button becomes "active" for a moment, which changes the text and icon, creating a brief flickering. This fixes it all.